application/src/test/java/org/opentripplanner/graph_builder/module/StreetLinkerModuleTest.java

```diff
@@ -103,6 +106,38 @@ void linkFlexStop() {
     });
   }
 
+  @Test
+  void linkFlexStopWithBoardingLocation() {
+    OTPFeature.FlexRouting.testOn(() -> {
+      var model = new TestModel().withStopLinkedToBoardingLocation();
+      var flexTrip = TimetableRepositoryForTest.of().unscheduledTrip("flex", model.stop());
+      model.withFlexTrip(flexTrip);
+
+      var module = model.streetLinkerModule();
+
+      module.buildGraph();
+
+      assertTrue(model.stopVertex().isConnectedToGraph());
+
+      // stop is used by a flex trip, needs to be linked to both the walk and car edge,
+      // also linked to the boarding location
+      assertThat(model.stopVertex().getOutgoing()).hasSize(3);
+      var links = model.outgoingLinks();
+      assertInstanceOf(BoardingLocationToStopLink.class, links.getFirst());
```

Comment:
The order of the links is not obvious, could either mention what the expected order is in a comment or test for the link type while iterating through the links?